src/Projection/InMemoryProjectionManager.php

```diff
@@ -0,0 +1,195 @@
+<?php
+/**
+ * This file is part of the prooph/event-store.
+ * (c) 2014-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2015-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Projection;
+
+use Prooph\EventStore\EventStore;
+use Prooph\EventStore\EventStoreDecorator;
+use Prooph\EventStore\Exception;
+use Prooph\EventStore\InMemoryEventStore;
+
+final class InMemoryProjectionManager implements ProjectionManager
+{
+    public const OPTION_CACHE_SIZE = 'cache_size';
+    public const OPTION_SLEEP = 'sleep';
+    public const OPTION_PERSIST_BLOCK_SIZE = 'persist_block_size';
+
+    private const DEFAULT_CACHE_SIZE = 1000;
+    private const DEFAULT_SLEEP = 100000;
+    private const DEFAULT_PERSIST_BLOCK_SIZE = 1000;
+
```

Comment:
@prolic There is a little issue, I don't know which option belongs to which `create*` method. For instance, `OPTION_PERSIST_BLOCK_SIZE` is only used in `createReadModelProjection()`. Maybe we should prefix the options, so you see for which method which option is available? An option without a prefix is available for all create methods?

Another solution is, the projection manager gets a factory for each type injected e.g. `QueryFactory`, `ReadModelProjectionFactory` and `ProjectionFactory` to decouple this stuff and put the options to each factory. The factory gets the event store on `__invoke`. /cc @oqq @codeliner